lib/model/model.go

```diff
@@ -1723,29 +1723,41 @@ func (m *Model) CheckFolderHealth(id string) error {
 	}
 
 	fi, err := os.Stat(folder.Path())
-	if v, ok := m.CurrentLocalVersion(id); ok && v > 0 {
-		// Safety check. If the cached index contains files but the
-		// folder doesn't exist, we have a problem. We would assume
-		// that all files have been deleted which might not be the case,
-		// so mark it as invalid instead.
-		if err != nil || !fi.IsDir() {
+
+	v, ok := m.CurrentLocalVersion(id)
+	indexHasFiles := ok && v > 0
+
+	if indexHasFiles {
+		// There are files in the folder according to the index, so it must
+		// have existed and had a correct marker at some point. Verify that
+		// this is still the case.
+
+		switch {
+		case err != nil || !fi.IsDir():
 			err = errors.New("folder path missing")
-		} else if !folder.HasMarker() {
+
+		case !folder.HasMarker():
 			err = errors.New("folder marker missing")
-		} else if free, errDfp := osutil.DiskFreePercentage(folder.Path()); errDfp == nil && free < folder.MinDiskFreePct {
-			err = errors.New("out of disk space")
+
+		case !folder.ReadOnly:
+			// Check for free space, if it isn't a master folder. We aren't
+			// going to change the contents of master folders, so we don't
+			// care about the amount of free space there.
+			if free, errDfp := osutil.DiskFreePercentage(folder.Path()); errDfp == nil && free < folder.MinDiskFreePct {
+				err = errors.New("out of disk space")
```

Comment:
No, but that's the error we had before as well. :) What should it say, you think? "free space reserve not met"?